### PR TITLE
Orders table implemented 

### LINKED ISSUE #7674: [PayPal] Implement the Orders table
### Short description of current behavior

MindsDB integrates with PayPal, so you can access PayPal data and make relevant predictions with MindsDB.

## Resources

Familiarize yourself with the existing PayPal integration: https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/paypal_handler

## Steps to Follow

Currently, the PayPal integration with MindsDB accesses the Payments table.

The task is to extend [this implementation](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/paypal_handler) to include the Orders table.

Please provide screenshots or a short video in the PR to show successful test results.

## :loudspeaker: Additional rewards :trophy:

Each integration extension issue brings 15+ points for SWAG and entry into the draw for a :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)

:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left: 

### Video or screenshots

_No response_

### Expected behavior

_No response_

### How to reproduce the error

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Tushar98644**
## Description

Implemented the orders table 

Fixes #7674 

## Type of change

(Please delete options that are not relevant)

- [] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] 📄 This change requires a documentation update

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.<br/>

https://github.com/mindsdb/mindsdb/assets/107763774/584c5891-1bf1-4170-9541-3b644b8dc73f

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/paypal_handler/paypal_tables.py [hunk 1]
```diff
@@ -82,3 +83,61 @@ def get_invoices(self, **kwargs) -> List[Dict]:
...
+    def get_columns(self,order_id) -> List[Text]:
+        return pd.json_normalize(self.get_orders(order_id)).columns.tolist()
+
+    def get_orders(self, order_id, **kwargs) -> List[Dict]:
+       headers = {
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer A21AAIUR5R-kPcdVNm6D6zNxxFlSFaE2smdAEZhGFAAsq92VGjrHYBjvZyMh4dzDklXjpDj3rL7vvDRmf5S4fHXcyK0PDwaVg',
+    }
+
+       response = requests.get(f'https://api-m.sandbox.paypal.com/v2/checkout/orders/{order_id}', headers=headers)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/paypal_handler/paypal_tables.py:131]: This should be alredy conected using the CREATE DATABASE statement. Check connection method

File: mindsdb/integrations/handlers/paypal_handler/paypal_tables.py [hunk 2]
```diff
@@ -82,3 +83,61 @@ def get_invoices(self, **kwargs) -> List[Dict]:
...
+
+        return orders_df
+    
+    def get_columns(self,order_id) -> List[Text]:
+        return pd.json_normalize(self.get_orders(order_id)).columns.tolist()
+
+    def get_orders(self, order_id, **kwargs) -> List[Dict]:
+       headers = {
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer A21AAIUR5R-kPcdVNm6D6zNxxFlSFaE2smdAEZhGFAAsq92VGjrHYBjvZyMh4dzDklXjpDj3rL7vvDRmf5S4fHXcyK0PDwaVg',
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/paypal_handler/paypal_tables.py:128]: This should be alredy conected using the CREATE DATABASE statement. Check connection method

File: mindsdb/integrations/handlers/paypal_handler/paypal_tables.py [hunk 3]
```diff
@@ -82,3 +83,61 @@ def get_invoices(self, **kwargs) -> List[Dict]:
...
+    }
+
+       response = requests.get(f'https://api-m.sandbox.paypal.com/v2/checkout/orders/{order_id}', headers=headers)
+
+       if response.status_code == 200:
+        data = response.json()
+        print(data)
+        return data
+       else:
+        print(f"Request failed with status code: {response.status_code}")
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/paypal_handler/paypal_tables.py:138]: Remove prints

File: mindsdb/integrations/handlers/paypal_handler/paypal_tables.py [hunk 4]
```diff
@@ -82,3 +83,61 @@ def get_invoices(self, **kwargs) -> List[Dict]:
...
+        print(f"Response content: {response.text}")
+        return []
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/paypal_handler/paypal_tables.py:140]: Why do we return empty array here?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

**Comment by ZoranPandovski** [comment]
@Tushar98644 The video doesn't show selecting Orders? By looking at the code this doesn't work and you have a lot of hardcoded values. Please check the comments above

**Comment by Tushar98644** [comment]
> @Tushar98644 The video doesn't show selecting Orders? By looking at the code this doesn't work and you have a lot of hardcoded values. Please check the comments above

@ZoranPandovski I am using the paypal order api as the paypalrestsdk has been depreciated and we may need to migrate to that in the near future , i am passing the endpoint and access_token in the query but facing this error : 

Screenshot 2023-10-17 at 9 47 26 PM

**Comment by Tushar98644** [comment]
I logged in a check for the `header` and `base_url` and it returns : `Both 'base_url' and 'headers' must be provided in connection_data.`

**Comment by Tushar98644** [comment]
@martyna-mindsdb @ZoranPandovski any updates ?

File: mindsdb/integrations/handlers/paypal_handler/paypal_tables.py [hunk 5]
```diff
@@ -100,6 +127,22 @@ def select(self, query: ast.Select) -> pd.DataFrame:
             where_conditions,
             order_by_conditions
         )
+        orders_df = select_statement_executor.execute_query()
```
Comment (martyna-mindsdb) [inline mindsdb/integrations/handlers/paypal_handler/paypal_tables.py:130]: This code (130-145 lines) should be under `class OrdersTable(APITable):` not here.

Please correct it and make sure it works.

Comment (martyna-mindsdb) [reply mindsdb/integrations/handlers/paypal_handler/paypal_tables.py:130]: There must have been some mistakes when merging the staging branch. Please address those.

Comment (Tushar98644) [reply mindsdb/integrations/handlers/paypal_handler/paypal_tables.py]: updated

**Comment by Tushar98644** [comment]
Screenshot 2023-10-22 at 6 31 16 PM

**Comment by martyna-mindsdb** [comment]
@Tushar98644 
Please address the comments that @ZoranPandovski left last week:

https://github.com/mindsdb/mindsdb/pull/7832#discussion_r1361837799
https://github.com/mindsdb/mindsdb/pull/7832#discussion_r1361837964

**Comment by Tushar98644** [comment]
> @Tushar98644 Please address the comments that @ZoranPandovski left last week:
> 
> [#7832 (comment)](https://github.com/mindsdb/mindsdb/pull/7832#discussion_r1361837799) [#7832 (comment)](https://github.com/mindsdb/mindsdb/pull/7832#discussion_r1361837964)

i have removed all the print statements , the handler is initiating the connection with the paypalrestsdk while i am using the paypalorders API , which requires just an authorization token, i don't really know how to go about this , do i need to to include it in the handler ?

**Comment by ZoranPandovski** [comment]
Please check other handlers, and see how they work and how we are passing the values from the CREATE DATABASE statment

**Comment by ZoranPandovski** [comment]
Closing this since it doesn't work. @Tushar98644 As discussed please check other handlers to see how this should work

**Pull request closed without merging**